plugin/health_check.go

```diff
@@ -12,105 +12,92 @@ import (
 )
 
 const (
-	HEALTH_CHECK_INTERVAL         = 30 * time.Second // How often the health check should run
-	HEALTH_CHECK_DISABLE_DURATION = 60 * time.Minute // How long we wait for num fails to incur before disabling the plugin
-	HEALTH_CHECK_PING_FAIL_LIMIT  = 3                // How many times we call RPC ping in a row before it is considered a failure
-	HEALTH_CHECK_RESTART_LIMIT    = 3                // How many times we restart a plugin before we disable it
+	HEALTH_CHECK_INTERVAL            = 30 * time.Second // How often the health check should run
+	HEALTH_CHECK_DEACTIVATION_WINDOW = 60 * time.Minute // How long we wait for num fails to incur before deactivating the plugin
+	HEALTH_CHECK_PING_FAIL_LIMIT     = 3                // How many times we call RPC ping in a row before it is considered a failure
+	HEALTH_CHECK_RESTART_LIMIT       = 3                // How many times we restart a plugin before we deactivate it
 )
 
-type PluginHealthCheckJob struct {
-	cancel     chan struct{}
-	cancelled  chan struct{}
-	cancelOnce sync.Once
-	env        *Environment
+type PluginHealthCheckStatus struct {
+	failTimestamps []time.Time
+	Crashed        bool
```

Comment:
if the healthchecker ends up setting the state on env, this might become unnecessary to track.